app/routes/create-catch-certificate/$catchCertificate/what-are-you-exporting/whatAreYouExporting.ts

```diff
@@ -130,20 +142,47 @@ export const getAddSpeciesLoaderData = async (catchCertificate: string, faoCode:
     getFavourites(),
     searchStateLookup(faoCode, stateCode),
     getCommodityCodes(faoCode, stateCode, presentationCode)
-  ])
+  ]);
 
   return {
     ...getAddedSpeciesPerUserData,
     species,
     favourites,
     stateLookup,
     commodityCodes,
-    faoCode,
+    ...findSpeciesName(faoCode, species),
     stateCode,
-    presentationCode
+    stateLabel: findStateLabel(stateCode, stateLookup),
+    presentationCode,
+    presentationLabel: findPresentationLabel(presentationCode, stateLookup),
   }
 }
 
+const findSpeciesName = (faoCode: string, species: ISpecies[]): ISpecies | undefined => {
+  if (!faoCode || !species.length) {
+    return undefined;
+  }
+
+  return species.find((species: ISpecies) => species.faoCode === faoCode)
+}
+
+
+const findStateLabel = (code: string, lookup: ISearchState): string | undefined  => {
+  if (!code || !lookup?.states) {
+    return undefined;
+  }
+
+  return lookup?.states.find((state: ILabelAndValue) => state.value === code)?.label
+}
+
+const findPresentationLabel = (presentationCode: string, lookup: ISearchState): string | undefined  => {
+  if (!presentationCode || !lookup?.presentations) {
+    return undefined;
+  }
+
+  return lookup?.presentations.find((presentation: ILabelAndValue) => presentation.value === presentationCode)?.label;
```

Comment:
I'm not 100% sure about it ... so is worth trying before. I think if lookup?.presentation is not present it return undefined already